pebbles/static/js/controllers/StatsController.js

```diff
@@ -2,9 +2,277 @@ app.controller('StatsController', ['$q', '$scope', '$http', '$interval', '$uibMo
                               function ($q,   $scope,   $http,   $interval,   $uibModal,   AuthService,   Restangular) {
 
         Restangular.setDefaultHeaders({token: AuthService.getToken()});
+        
+        $scope.currentView = "usageView";
+
+        $scope.formView;
+
+        $scope.toggleClass = function(view){
+            if ($scope.currentView == view){
+                $scope.formView = view;
+                return "active";
+            }
+            return undefined;
+        }
+        $scope.isCurrentView = function(view){
+            if ($scope.currentView == view){
+                return true;
+            }
+            return false;
+        }
+
+        $scope.totalItems;
+        $scope.visiblePages = 4;
+        $scope.currentPage = 1;
+        var itemsPerPage = 15;
+        $scope.itemsPerPage = itemsPerPage;
+
+        $scope.userlist;
+        $scope.dates = {};
+
+
+        var exportInstitutions = Restangular.all('export_stats/institutions');
+        var exportActiveUsers = Restangular.all('export_stats/active_users');
+        var exportMonthlyInstances = Restangular.all('export_stats/monthly_instances');
+        var exportQuartals = Restangular.all('export_stats/quartals');
+        var exportQuartalsByOrganization = Restangular.all('export_stats/quartals_by_organization')
+
         var stats = Restangular.oneUrl('stats');
+        var activeUsers = Restangular.oneUrl('export_stats/active_users');
+        var inst = Restangular.oneUrl('export_stats/institutions');
+        var monthlyInstances = Restangular.oneUrl('export_stats/monthly_instances');
+        var quartals = Restangular.oneUrl('export_stats/quartals');
+        var quartalsByOrganization = Restangular.oneUrl('export_stats/quartals_by_organization');
+
 
         stats.get().then(function (response) {
             $scope.stats = response;
+        });    
+
+        activeUsers.get($scope.dates).then(function(response){  
+            $scope.activeUsers = response;
+            $scope.totalItems = $scope.activeUsers[0].data.length;
+            $scope.pagination($scope.currentPage);
+        });
+
+        inst.get($scope.dates).then(function (response) {
+            $scope.inst = response;
+        });
+
+        monthlyInstances.get($scope.dates).then(function(response) {
+            $scope.monthlyInstances = response;
+        });
+
+        quartals.get($scope.dates).then(function(response) {
+            $scope.quartals = response;
+        });
+
+        quartalsByOrganization.get($scope.dates).then(function(response) {
+            $scope.quartalsByOrganization = response;   
+                 
+        });
+
+        $scope.pagination = function(page) {
+            var data = $scope.activeUsers[0].data.slice(
+                (page-1) * $scope.itemsPerPage,
+                page * $scope.itemsPerPage
+            );
+            $scope.userlist = data;
+        };
```

Comment:
The whole users list is fetched every single time (for every page numbers, every loading/reloads) and then slicing is done in frontend. It is not efficient when the number of users are real high. Do the slicing at the backend and fetch them here. This feature is already here [(https://github.com/CSCfi/pebbles/commit/ca2ff8b6d1ad97849afa1f61ada77323aba637c4)](https://github.com/CSCfi/pebbles/commit/ca2ff8b6d1ad97849afa1f61ada77323aba637c4) you just have to call it.